[CARBONDATA-708] Fixed Between Filter Issue for Pushed Filters in case it was applied …

bhavya411 opened the pull request:
Fixed Between Filter Issue for Pushed Filters in case it was applied on Dimension
Test
Add Examples of Between Filter
Removed Scalastyle 
Manually Test BETWEEN Filter
Run Unit Test Cases

CarbonDataQA commented:
Can one of the admins verify this patch?

chenliang613 commented:
add to whitelist

CarbonDataQA commented:
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/982/

CarbonDataQA commented:
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/985/

CarbonDataQA commented:
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/993/

CarbonDataQA commented:
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1015/

ravipesala commented:
Please rebase the code.

ravipesala commented on integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala:
```diff
@@ -110,8 +110,16 @@ object CarbonFilters {
+      val dataTypeOfAttribute = CarbonScalaUtil.convertSparkToCarbonDataType(dataTypeOf(name))
+      val dataType = if (Option(value).isDefined
+        && dataTypeOfAttribute == DataType.STRING
+        && value.isInstanceOf[Double]) {
```
Can you explain with out this fix what is the issue happening?

bhavya411 replied:
In the scenarios where we have stored the numbers as String (and have included it as a Dimension) and if we are trying to do operations like >= or <= or between it is not working correctly 
and it gives some random results as it is comparing a String with a Double. So for those cases this fix will work.

bhavya411 commented:
I have rebased the code

CarbonDataQA commented:
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1139/

ravipesala commented on examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSelectFilterExample.scala:
```diff
@@ -0,0 +1,104 @@
+
+
+
+object CarbonSelectFilterExample {
```
Please move this to common-test

ravipesala commented:
@bhavya411 thanks for working on this.  LGTM but please move the examples to testcase.

CarbonDataQA commented:
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1304/

bhavya411 commented:
@ravipesala Created a New PR #697 for the fixes that were suggested , added a Unit test class to make sure it runs everytime with the build and removed the example
